app/src/main/java/com/borconi/emil/wifilauncherforhur/listeners/WifiListener.java

```diff
@@ -0,0 +1,209 @@
+
+        if (connectivityManager != null) {
+
+
+            NetworkRequest.Builder builder = new NetworkRequest.Builder();
+            builder.addTransportType(NetworkCapabilities.TRANSPORT_WIFI);
+            networkCallback = new ConnectivityManager.NetworkCallback() {
+                @Override
+                public void onLost(Network network) {
+                    Log.d("Wifi Listener", "Lost connection to HUR wifi, exiting the app");
+                    connectivityManager.unregisterNetworkCallback(this);
+                    stopSelf();
+                }
+            };
+            connectivityManager.registerNetworkCallback(
+                    builder.build(), networkCallback
+            );
+        }
+
+    }
+
+    @Override
+    public void startSlave(final WifiP2pInfo info, final WifiP2pGroup wifiP2pGroup) {
+        if (hur_address != null) {
+            return;
+        }
+        hur_address = info.groupOwnerAddress;
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    Log.d("HU-Wifip2p", "Trying to connect to: " + info.groupOwnerAddress);
+                    Socket sock = new Socket(info.groupOwnerAddress, 5299);
+                    InputStream inpstr = sock.getInputStream();
+                    while (inpstr.available() <= 0) {
+                        Thread.sleep(10);
+                    }
+                    byte[] pass = new byte[inpstr.available()];
+                    inpstr.read(pass);
+                    sock.close();
+                    String password = new String(pass);
+                    Log.d("HU-Wifip2p", "Password from socket: " + password);
+                    WifiConfiguration wifiConfig = new WifiConfiguration();
+                    wifiConfig.SSID = String.format("\"%s\"", wifiP2pGroup.getNetworkName());
+                    wifiConfig.preSharedKey = String.format("\"%s\"", password);
+                    mManager.removeGroup(mChannel, null);
+                    WifiManager wifiManager = (WifiManager) getApplicationContext().getSystemService(WIFI_SERVICE);
+                    assert wifiManager != null;
+                    netId = wifiManager.addNetwork(wifiConfig);
+                    mManager.removeGroup(mChannel, null);
+                    wifiManager.disconnect();
+                    wifiManager.enableNetwork(netId, true);
+                    wifiManager.reconnect();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        }).start();
+    }
+
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        Log.d("WiFi-Launcher", "Start service");
+        return START_STICKY;
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+
+        isConnected = false;
```

Comment:
This is really important here. I would like to refactor it to avoid this kinda of static properties, the bug was entirely logical